shapeAnalysisProject/dimension_reduction/PCA.py

```diff
@@ -3,7 +3,7 @@
 from sklearn.decomposition import PCA
 import matplotlib.pyplot as plt
 
-def show_plot(similarity, filenames=[]):
+def show_plot(value, title, similarity, filenames=[]):
```

Comment:
i would write the function as:
`def show_plot(similarity, filenames=[], value=False, title='')`
That way it is not mandatory to pass `value` or `title` and also, the initial order of arguments of `show_plot` remain same.

Please maintain the order in which the arguments are already defined. We should try to avoid changing function definition but as it seems, it is required here, so add new parameters at the end of function definition and **not** in the beginning.

Reply:
1. Value and title for the plots are different so i had to take variables to assign different values for the two plots.
2. The order of arguments had to be changed to avoid syntax error : non-static arguments cannot be passed before static arguments

Reply:
but why do we want to add static arguments? `value` could be predefined to `False` and `title=''`

Reply:
Here, static arguments to these variables have been passed in the file main.py where this function has been called, which need to be passed even if they are predefined in this function show_plot(). Nevertheless i will do the changes check them and commit accordingly if required.